tasty-inspector/src/scala/tasty/inspector/TastyInspector.scala

```diff
@@ -57,13 +58,39 @@ trait TastyInspector:
     val files = tastyFiles ::: jars
     files.nonEmpty && inspectFiles(dependenciesClasspath, files)
 
-  private def inspectFiles(classpath: List[String], classes: List[String]): Boolean =
-    if (classes.isEmpty)
-      throw new IllegalArgumentException("Parameter classes should no be empty")
+  /** Load and process TASTy files using TASTy reflect and provided context
+   *
+   *  Used in doctool to reuse reporter and setup provided by sbt
+   *
+   *  @param classes List of paths of `.tasty` and `.jar` files (no validation is performed)
+   *  @param classpath Classpath with extra dependencies needed to load class in the `.tasty` files
+   */
+  protected def inspectFilesInContext(classpath: List[String], classes: List[String])(using Context): Unit =
```

Comment:
I think we should hide this definition from the user-facing API, having it here just opens a can of worms we'd prefer to have closed.

How about:
1) we make `inspectorDriver` `private[inspector]`
2) we create `class DocTastyInspector extends TastyInspector` in pkg `scala.tasty.inspector.internal`
3) we move this definition to said new class
4) we use that class in the doctool instead of the usual inspector

Reply:
Added in: https://github.com/lampepfl/dotty/pull/10462/commits/2cd20e42c06f62fe4f4e468585cdcd90be22de77